causal_testing/testing/causal_test_outcome.py

```diff
@@ -99,10 +95,9 @@ def apply(self, res: CausalTestResult) -> bool:
         if res.ci_valid() and not super().apply(res):
             return False
         if res.test_value.type in {"ate", "coefficient"}:
-            return bool(res.test_value.value > 0)
+            return bool(res.test_value.value[0] > 0)
```

Comment:
Again, this breaks scalability to categorical data types, although this one is a bit more complicated. For a "positive" effect, you need an order. For now, we could just support float datatypes. To make sure we're not losing any information, we could raise a `TypeError` if `len(res.test_value.value) > 1`?

Reply:
For a categorical it isn't enough to just check that every variable is positive (greater than 0)? 

The suggestion of raising an error seems very logical to me to avoid unexpected behaviour though for now

Reply:
Does this not just check the first one though? I'd have expected to see an `all` in there somewhere...

Reply:
Sorry should have been clearer! It does currently only check the first one, but if I were to change it to check that all values are positive would that be enough?

I'm not sure what you meant by 'you need an order'.

Reply:
In this case, I don't think so. A positive relationship here means "as we increase treatment X, we expect to see a corresponding increase in outcome Y". For this to apply, we need the value of X to be orderable such that "increasing X" is meaningful. If we just assert that all values are positive, we are just asserting that whatever value statsmodels uses as its reference value is the one for which we observe the minimal outcome value (and thus that all other treatment values would lead to a higher outcome). For now, I think it's better to just assert that there's only one value and then just index it. Perhaps we could add this explanation as part of the documentation for posterity.